devolving_music/lib/song_scores.py

```diff
@@ -29,18 +31,25 @@ def get_scores(self) -> "dict[int, ScoreSuite]":
 
         return self.song_score_dict
 
-    def get_compare_submission_random(self, submission_id) -> SongSubmission:
+    def get_scores_list(self) -> "list[ScoreSuite]":
+        # calculates scores for all submissions using current comparisons
+        # returns list of all score suite objects
+        return list(self.get_scores().values())
+
+    def get_compare_submission_random(self, submission_id , remove_submission = True) -> SongSubmission:
         key_list = list(self.song_score_dict.keys())
-        key_list.remove(submission_id)
+        if (remove_submission):
+            key_list.remove(submission_id)
         # once you have a critical number of comparisons then pull from quality
         # list
-        return self.song_score_dict.get(random.choice(key_list)).song_submission
+        return self.song_score_dict.get(
+            random.choice(key_list)).song_submission
```

Comment:
Getting a random song submission and getting a song submission out of some restricted set with the intention of comparing it to something are different enough actions that I think they're clearer as different methods. Notice that, as written, you're forced to provide a `submission_id` argument even though it goes ignored unless `remove_submission` is true - that's a sure sign the API design could use some more work.

I think there are a few reasonable solutions here. How about something like:

```python
def get_compare_submission_random(self, submission_id: int):
  return self.get_random_submission(excluded_ids={submission_id})

def get_random_submission(self, excluded_ids: Optional[set[int]] = None) -> SongSubmission:
  keys = set(self.song_score_dict.keys())
  if excluded_ids:
    keys -= excluded_ids

  return self.song_score_dict.get(
            random.choice(keys)).song_submission

Reply:
Eh it's not that big of a need it was just an option that I thought could be useful. I'll just get rid of the option and have it as it was before where it'll always remove the key. I think this overcomplicates the issue, which is honestly a nonissue.